src/gs-update-monitor.c

```diff
@@ -200,6 +512,12 @@ get_updates_finished_cb (GObject *object,
 	    no_updates_for_a_week (monitor)) {
 		notify_offline_update_available (monitor);
 	}
+
+	/* try to update any apps that should be updated already, and connect to
+	 * MwscScheduleEntry signals; this is done here because now all the updates
+	 * have been scheduled, otherwise we would risk starting an update only
+	 * for it to be canceled if a higher priority app was added */
+	finish_scheduling_updates (monitor);
```

Comment:
You'll have to move this somewhere else if you implement `schedule_update` in an asynchronous way (instead of manually spinning the main loop, as you do now)